src/kit/RadioGroup.tsx

```diff
@@ -0,0 +1,157 @@
+import { Radio } from 'antd';
+import { RadioChangeEvent } from 'antd/lib/radio';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { throttle } from 'throttle-debounce';
+
+import Icon, { IconName, IconSize } from 'kit/Icon';
+import useResize from 'kit/internal/useResize';
+import Tooltip from 'kit/Tooltip';
+
+import css from './RadioGroup.module.scss';
+
+export const DEFAULT_RESIZE_THROTTLE_TIME = 500;
+
+interface Props {
+  defaultValue?: string;
+  iconOnly?: boolean;
+  onChange?: (id: string) => void;
+  options: RadioGroupOption[];
+  value?: string;
+  radioType?: 'button' | 'radio';
+}
+
+export interface RadioGroupOption {
+  icon?: IconName;
+  iconSize?: IconSize;
+  id: string;
+  label: string;
+}
+
+interface SizeInfo {
+  baseHeight: number;
+  baseWidth: number;
+  parentWidth: number;
+}
+
+const RESIZE_THROTTLE_TIME = 500;
+const PARENT_WIDTH_BUFFER = 16;
+const HEIGHT_LIMIT = 50;
+
+const RadioGroup: React.FC<Props> = ({
+  defaultValue,
+  iconOnly = false,
+  onChange,
+  options,
+  value,
+  radioType = 'button',
+}: Props) => {
+  const baseRef = useRef<HTMLDivElement>(null);
+  const originalWidth = useRef<number>();
+  const resize = useResize();
+  const [showLabels, setShowLabels] = useState(true);
+  const [sizes, setSizes] = useState<SizeInfo>({ baseHeight: 0, baseWidth: 0, parentWidth: 0 });
+  const classes = [css.base];
+
+  const hasIconsAndLabels = useMemo(() => {
+    if (options.length === 0) return false;
+    return options.reduce((acc, option) => acc || (!!option.icon && !!option.label), false);
+  }, [options]);
+
+  if (iconOnly) classes.push(css.iconOnly);
+
+  const handleChange = useCallback(
+    (e: RadioChangeEvent) => {
+      if (onChange) onChange(e.target.value);
+    },
+    [onChange],
+  );
+
+  /*
+   * Dynamically check to see if labels can be shown along with the icons,
+   * if there's room to show both and both are available.
+   */
+  useEffect(() => {
+    if (!hasIconsAndLabels || !baseRef.current) return;
+    if (sizes.baseWidth === 0 || sizes.parentWidth === 0) return;
+
+    setShowLabels((prev) => {
+      if (!originalWidth.current) return prev;
+      if (prev && sizes.baseHeight > HEIGHT_LIMIT) return false;
+      if (!prev && originalWidth.current < sizes.parentWidth) return true;
+      return prev;
+    });
+  }, [hasIconsAndLabels, sizes]);
+
+  /*
+   * Update parent and component sizes upon resize of the window,
+   * at a throttled rate.
+   */
+  useEffect(() => {
+    const throttleFunc = throttle(RESIZE_THROTTLE_TIME, () => {
```

Comment:
we should be able to use the ref we get from the `useResize` handler to get the baseHeight and baseWidth.

Reply:
`useResize` returns an object containing a `refCallback` you can pass into an element's `ref` property to have it listen to that instead of `document.body`. It also returns a `ref` we can use here to get the parent.